.cursor/rules/api-specific-config.mdc

```diff
@@ -10,11 +10,128 @@ alwaysApply: false
 ### Database Schema Organization
 - All database schemas must be placed in `packages/db-schema/src/` with appropriate subdirectories (`core/`, `auth/`, etc.)
 - Schema files must export table definitions and be properly imported in `core/defs.ts`
-- Always generate migrations using `pnpm db:generate` after schema changes which will automatically create the sql migration files
+- Always generate migrations using `pnpm db:gen-migrations` after schema changes which will automatically create the sql migration files
 - Repository classes must be placed in `apps/api/src/database/repositories/` and follow the naming pattern `*-repository.ts`
 - Use the established repository pattern with dependency injection into manager services
 - Include proper indexes for foreign keys and frequently queried columns
 
+### Database Computation Patterns
+- **SQL-First Philosophy**: Always prefer database aggregations over in-memory processing
+  - Use SQL `SUM()`, `AVG()`, `COUNT()` instead of fetching all rows and calculating in JavaScript
+  - Use `DISTINCT ON` with proper indexes for latest-record queries
+  - Use database-level `GROUP BY` for grouping operations
+  - Use CTEs (Common Table Expressions) for complex multi-step queries
+  - Use `ORDER BY` in SQL instead of JavaScript `.sort()`
+  - For complex sorts (like `DISTINCT ON` + different `ORDER BY`), use subqueries with `.as()` in Drizzle
+  - Example: Getting latest summaries should use `DISTINCT ON (agent_id)` not fetch all and filter in memory
+- **Response Size Management**:
+  - Never return unbounded result sets for in-memory processing
+  - Always aggregate at database level when only totals are needed
+  - Example: Use `SUM(total_volume)` in SQL vs fetching 1000+ records to sum in JavaScript
+  - Set reasonable LIMIT clauses for safety even on internal queries
+  - Monitor query response sizes in production
+- **Composite Indexes**: Always add indexes for:
+  - Foreign key relationships
+  - Columns used in `WHERE` clauses
+  - Columns used in `ORDER BY` with `DISTINCT ON`
+  - Multi-column patterns that match query patterns (e.g., `(agent_id, competition_id, timestamp DESC)`)
+- **Query Optimization Principles**:
+  - Prevent the N+1 query problem: Instead of fetching a list then making separate queries per item, use joins or batch fetching
+  - Prevent the linear query scaling problem: Instead of fetching a list then making separate queries per item, use joins or batch fetching
```

Comment:
the suggestion was for this line to *replace* the one above.  I'd like to remove all references to the "N+1 query problem"